apps/website/src/pages/clips/index.tsx

```diff
@@ -0,0 +1,330 @@
+import { PageNavigation } from "@/components/shared/PageNavigation";
+
+import leafLeftImage2 from "@/assets/floral/leaf-left-2.png";
+import leafRightImage2 from "@/assets/floral/leaf-right-2.png";
+import { trpc } from "@/utils/trpc";
+import type { SortType } from "@/server/db/clips";
+import { Button } from "@/components/shared/Button";
+import IconLoading from "@/icons/IconLoading";
+import Consent from "@/components/Consent";
+import IconThumbsUp from "@/icons/IconThumbsUp";
+import { classes } from "@/utils/classes";
+import DateTime from "@/components/content/DateTime";
+
+type Clip = {
+  id: string;
+  slug: string;
+  title: string;
+  voteCount: number;
+  thumbnailUrl: string;
+  hasVoted: boolean;
+  createdAt: Date;
+  creator: string | null;
+};
+
+const ClipItem = ({
+  clip,
+  clipClicked,
+  lastItemRef,
+}: {
+  clip: Clip;
+  clipClicked: (clip: Clip) => void;
+  lastItemRef: React.RefObject<HTMLDivElement>;
+}) => {
+  const { data: session } = useSession();
+
+  const addVote = trpc.clips.addVote.useMutation({
+    onSuccess: () => {
+      setHasVoted(true);
+      clip.voteCount++;
+    },
+  });
+
+  const removeVote = trpc.clips.removeVote.useMutation({
+    onSuccess: () => {
+      setHasVoted(false);
+      clip.voteCount--;
+    },
+  });
+
+  const handleVote = (clipId: string) => {
+    if (hasVoted) {
+      removeVote.mutate({ clipId });
+    } else {
+      addVote.mutate({ clipId });
+    }
+  };
+
+  const [hasVoted, setHasVoted] = useState(clip.hasVoted);
+  const thumbsClass = hasVoted
+    ? "hover:fill-alveus-green-900 fill-green"
+    : "hover:fill-green fill:alveus-green-900";
+
+  return (
+    <div>
+      <div
+        ref={lastItemRef}
+        className="group cursor-pointer"
+        onClick={() => clipClicked(clip)}
+      >
+        <Image
+          src={clip.thumbnailUrl}
+          alt=""
+          width={700}
+          height={300}
+          className="h-auto w-full rounded-xl transition group-hover:scale-102 group-hover:shadow-lg"
+        />
+        <div className="min-h-[4rem]">
+          <p className="text-center text-xl font-medium transition-colors group-hover:text-alveus-green-700 ">
+            {clip.title}
+          </p>
+        </div>
+      </div>
+      <div className="flex justify-between px-4">
+        <p className="text-left">Votes: {clip.voteCount}</p>
+        {session && (
+          <p
+            className={classes("cursor-pointer text-right", thumbsClass)}
+            onClick={() => handleVote(clip.id)}
+          >
+            <IconThumbsUp size={20} />
+          </p>
+        )}
+      </div>
+    </div>
+  );
+};
+
+const ClipModal = ({
+  clip,
+  onClose,
+}: {
+  clip: Clip;
+  onClose: (e: React.MouseEvent) => void;
+}) => {
+  const clipUrl = new URL("https://clips.twitch.tv/embed");
+  clipUrl.searchParams.set("clip", clip.slug);
+  clipUrl.searchParams.set("parent", window.location.hostname);
+
+  return (
+    <div
```

Comment:
The modal dialog should capture keyboard focus and close when pressing the escape key and have appropriate aria attributes. The easiest would be to use the headless `Dialog` component (see https://github.com/alveusgg/alveusgg/blob/main/apps/website/src/components/shared/ModalDialog.tsx). This will require some tweaks to the event handler but it should be fairly easy to replace the divs with `Dialog`, `Dialog.Panel` and the headline with a `Dialog.Title`. Reusing the existing `ModalDialog` might also work but the button is fixed as a labeled button at the bottom of the dialog ...